src/routes/webhooks.js

```diff
@@ -0,0 +1,201 @@
+      const response = await collection.updateOne(offer, {
+        $set: {
+          min: req.body._lowerLimitFn ? req.body._lowerLimitFn : offer.min,
+        },
+      });
+      if (response.modifiedCount > 0)
+        trigger('offer', { offerId: req.body._idOffer });
+      return res.status(200).send(response);
+    }
+    res.status(400).send({
+      msg: 'Not found.',
+    });
+  }
+);
+
+/* This is a PUT request that updates token offer. */
+router.put('/offer/token', updateTokenOfferValidator, async (req, res) => {
+  const validator = validateResult(req, res);
+  const collection = (await getDBConnection(req)).collection('offers');
+  if (validator.length) {
+    return res.status(400).send(validator);
+  }
+  const offer = await collection.findOne({
+    offerId: req.body._idOffer,
+  });
+  if (offer) {
+    const response = await collection.updateOne(offer, {
+      $set: {
+        tokenAddress: req.body._token ? req.body._token : offer.tokenAddress,
+      },
+    });
+    if (response.modifiedCount > 0)
+      trigger('offer', { offerId: req.body._idOffer });
+    return res.status(200).send(response);
+  }
+  res.status(400).send({
+    msg: 'Not found.',
+  });
+});
+
+/* This is a PUT request that updates chain id offer. */
+router.put('/offer/chain', updateChainIdOfferValidator, async (req, res) => {
+  const validator = validateResult(req, res);
+  const collection = (await getDBConnection(req)).collection('offers');
+  if (validator.length) {
+    return res.status(400).send(validator);
+  }
+  const offer = await collection.findOne({
+    offerId: req.body._idOffer,
+  });
+  if (offer) {
+    const response = await collection.updateOne(offer, {
+      $set: {
+        chainId: req.body._chainId ? req.body._chainId : offer.chainId,
+      },
+    });
+    if (response.modifiedCount > 0)
+      trigger('offer', { offerId: req.body._idOffer });
+    return res.status(200).send(response);
+  }
+  res.status(400).send({
+    msg: 'Not found.',
+  });
+});
+
+/* This is a PUT request that updates status offer. */
+router.put('/offer/status', updateStatusOfferValidator, async (req, res) => {
+  const validator = validateResult(req, res);
+  const collection = (await getDBConnection(req)).collection('offers');
+  if (validator.length) {
+    return res.status(400).send(validator);
+  }
+  const offer = await collection.findOne({
+    offerId: req.body._idOffer,
+  });
+  if (offer) {
+    const response = await collection.updateOne(offer, {
+      $set: {
+        isActive:
+          req.body._isActive === undefined
+            ? offer.isActive
+            : req.body._isActive,
+      },
+    });
+    if (response.modifiedCount > 0)
+      trigger('offer', { offerId: req.body._idOffer });
+    return res.status(200).send(response);
+  }
+  res.status(404).send({
+    msg: 'Not found.',
+  });
+});
+
+/* This is a PUT request that updates new offer. */
+router.put('/offer', updateOfferValidator, async (req, res) => {
```

Comment:
> @tcoratger
> 
> How should we proceed with this [event](https://github.com/grindery-io/Depay-Reality/blob/2b0596f883ac32bf0338a05e7f71e027ab32f8d6/contracts/v-testnet-launch/GrtPool.sol#L83) and this [one](https://github.com/grindery-io/Depay-Reality/blob/2b0596f883ac32bf0338a05e7f71e027ab32f8d6/contracts/v-testnet-launch/GrtOffer.sol#L81) if we do not have every param needed to update the database? Few of the params only the frontend knows.
> 
> For LogNewOffer, it's the same but on the Gordon side this time. The front-end will provide the hash of the transaction before it is completed, then you just have to fill in the missing elements in the offers collection of the database, corresponding to the hash of the transaction you receive of the blockchain and which should already be stored in the database.

For this [event](https://github.com/grindery-io/Depay-Reality/blob/2b0596f883ac32bf0338a05e7f71e027ab32f8d6/contracts/v-testnet-launch/GrtOffer.sol#L81) looks like every param is already filled in db, so it's not being updated any doc.

This is the payload for the event listened:

> {
>   _chainId: '97',
>   _grinderyChainId: '5',
>   _grinderyTransactionHash: '0x5d9b45f28bd37ebe3afb0f9002fa244ad5e359dee8218eca02337a22d8b5e758',
>   _idOffer: '0xcb21b0170c84ded985587ebf1cf042acd5aba5407914a30a872f1f0b97048767',
>   _token: '0x0000000000000000000000000000000000000000'
> }

Reply:
@juniormp For this event, it is Gordon who has just set up an offer, he clicked to confirm on Metamask: we were therefore able to capture his hash of the transaction. 

But we don't have the `offerId` yet (which we could theoretically calculate off-chain but not with a 100% confidence), but we prefer to wait for the response from the blockchain. 

So, in the same way as before, you listen to this event, you search the offers database for the hash and you fill in with the `offerId` which is normally empty for the moment.